tests/delegated_scanning_test.go

```diff
@@ -942,20 +919,40 @@ func (ts *DelegatedScanningSuite) resetConfig(ctx context.Context) {
 func (ts *DelegatedScanningSuite) setConfig(ctx context.Context, image deleScanTestImage) {
 	t := ts.T()
 
-	conn := centralgrpc.GRPCConnectionToCentral(t)
-	service := v1.NewDelegatedRegistryConfigServiceClient(conn)
-
 	cfg := &v1.DelegatedRegistryConfig{
 		EnabledFor:       v1.DelegatedRegistryConfig_SPECIFIC,
 		DefaultClusterId: ts.remoteCluster.GetId(),
 		Registries: []*v1.DelegatedRegistryConfig_DelegatedRegistry{
 			{Path: image.Base()},
 		},
 	}
-	_, err := service.UpdateConfig(ctx, cfg)
+
+	err := ts.updateConfigWithRetries(ctx, cfg)
 	require.NoError(t, err)
 }
 
+// updateConfigWithRetries will update the delegated registry config in Central
+// retrying on connection refused error. This was added to reduce flakes in
+// CI.
+func (ts *DelegatedScanningSuite) updateConfigWithRetries(ctx context.Context, cfg *v1.DelegatedRegistryConfig) error {
+	t := ts.T()
+
+	conn := centralgrpc.GRPCConnectionToCentral(t)
+	service := v1.NewDelegatedRegistryConfigServiceClient(conn)
+
+	retryFunc := func() error {
+		_, err := service.UpdateConfig(ctx, cfg)
+
+		if err != nil && strings.Contains(err.Error(), "connection refused") {
```

Comment:
Can you please extract a function that looks for this string and mangles `err` and reuse it below?